packages/inventories-exporter/src/main.js

```diff
@@ -0,0 +1,218 @@
+    }
+  }
+
+  _fetchInventories (
+    outputStream: stream$Writable,
+  ): Promise<any> {
+    if (this.exportConfig.channelKey)
+      return this._resolveChannelKey(this.exportConfig.channelKey)
+        .then(channelId => this._makeRequest(outputStream, channelId))
+    return this._makeRequest(outputStream)
+  }
+
+  _makeRequest (outputStream: stream$Writable, channelId?: string) {
+    const query = this.reqBuilder
+      .inventory
+      .expand('custom.type')
+      .expand('supplyChannel')
+      .whereOperator('and')
+    if (this.exportConfig.queryString)
+      query.where(this.exportConfig.queryString)
+    if (channelId)
+      query.where(`supplyChannel(id="${channelId}")`)
+    const uri = query.build()
+    const request: ClientRequest = {
+      uri,
+      method: 'GET',
+    }
+    if (this.accessToken)
+      request.headers = {
+        Authorization: `Bearer ${this.accessToken}`,
+      }
+    return this.client.process(
+      request,
+      (payload: Object): Promise<any> => InventoryExporter._processFn(
+        payload.body.results,
+        outputStream,
+      ),
+      { accumulate: false },
+    )
+  }
+
+  _resolveChannelKey (channelKey: string): Promise<any> {
+    const queryString = `key="${channelKey}"`
+    const uri = this.reqBuilder
+      .channels
+      .where(queryString)
+      .build()
+    const request: ClientRequest = {
+      uri,
+      method: 'GET',
+    }
+    if (this.accessToken)
+      request.headers = {
+        Authorization: `Bearer ${this.accessToken}`,
+      }
+    return this.client.execute(request)
+      .then((result): Promise<any> => {
+        if (result.body && result.body.results.length)
+          return Promise.resolve(result.body.results[0].id)
+        return Promise.reject(
+          new Error('No data with channel key in CTP Platform'),
+        )
+      })
+  }
+
+  static _processFn (
+    inventories: Array<Inventory>,
+    outputStream: stream$Writable,
+  ): Promise<any> {
+    InventoryExporter._writeEachInventory(outputStream, inventories)
+    return Promise.resolve()
+  }
+  // map to format acceptable by csv especially for import
+  static inventoryMappings (row: Inventory): CsvInventoryMapping {
+    const result: CsvInventoryMapping = {
+      sku: row.sku,
+      quantityOnStock: row.quantityOnStock,
+    }
+    if (row.supplyChannel && row.supplyChannel.obj)
+      result.supplyChannel = row.supplyChannel.obj.key
+    if (row.restockableInDays)
+      result.restockableInDays = row.restockableInDays
+    if (row.expectedDelivery)
+      result.expectedDelivery = row.expectedDelivery
+    if (row.custom && Object.keys(row.custom).length !== 0) {
+      const customObj = row.custom
+      result.customType = customObj.type.obj.key
+      const keys = Object.keys(customObj.fields)
+      keys.forEach((key: string) => {
+        result[`custom.${key}`] = customObj.fields[key]
+      })
+    }
+    return result
+  }
+
+  static _writeEachInventory (
```

Comment:
Hmmm, a lot of refactoring when down the line

but yeah, I think no much benefit of this